examples/bcauss.py

```diff
@@ -0,0 +1,124 @@
+args = parser.parse_args()
+
+path = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data', "jobs")
+dataset = Jobs(root=path, feature_engineering=args.feature_engineering)
+ATT = dataset.get_att()
+print(f"ATT is {ATT}")
+
+torch.manual_seed(args.seed)
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+dataset.materialize(path=osp.join(path, "data.pt"))
+
+dataset = dataset.shuffle()
+"""
+if dataset.split_col is None:
+    train_dataset, val_dataset, test_dataset = dataset[:0.62], dataset[
+    0.62:0.80], dataset[0.80:]
+else:
+    train_dataset, val_dataset, test_dataset = dataset.split()
+"""
+train_dataset = dataset
+treated_df = dataset.df[(dataset.df['source'] == 1)
+                        & (dataset.df['treated'] == 1)]
+treated_eval_dataset = Dataset(treated_df, dataset.col_to_stype,
+                               target_col='target')
+control_df = copy.deepcopy(treated_df)
+control_df['treated'] = 0
+control_eval_dataset = Dataset(control_df, dataset.col_to_stype,
+                               target_col='target')
+
+treated_eval_dataset.materialize(path=osp.join(path, "treated_eval_data.pt"))
+control_eval_dataset.materialize(path=osp.join(path, "control_eval_data.pt"))
+
+train_tensor_frame = train_dataset.tensor_frame
+treatment_idx = train_tensor_frame.col_names_dict[stype.categorical].index(
+    'treated')
+# val_tensor_frame = val_dataset.tensor_frame
+# test_tensor_frame = test_dataset.tensor_frame
+treated_eval_tensor_frame = treated_eval_dataset.tensor_frame
+# This is a bad hack. Currently the materialization logic would override 1's
+# to 0's due to 1's being the popular class
+treated_eval_tensor_frame.feat_dict[stype.categorical][:, treatment_idx] = 1.
+control_eval_tensor_frame = control_eval_dataset.tensor_frame
+
+train_loader = DataLoader(train_tensor_frame, batch_size=args.batch_size,
+                          shuffle=True)
+# val_loader = DataLoader(val_tensor_frame, batch_size=args.batch_size)
+# test_loader = DataLoader(test_tensor_frame, batch_size=args.batch_size)
+
+model = BCAUSS(
+    channels=train_tensor_frame.num_cols - 1,
+    hidden_channels=200,
+    decoder_hidden_channels=100,
+    out_channels=1,
+    col_stats=dataset.col_stats if not args.feature_engineering else None,
+    col_names_dict=train_tensor_frame.col_names_dict
+    if not args.feature_engineering else None,
+).to(device)
+
+optimizer = torch.optim.SGD(model.parameters(), lr=args.lr)
+lr_scheduler = ExponentialLR(optimizer, gamma=0.95)
+
+is_classification = True
+
+
+def train(epoch: int) -> float:
+    model.train()
+    loss_accum = total_count = 0
+
+    for tf in tqdm(train_loader, desc=f'Epoch: {epoch}'):
+        tf = tf.to(device)
+        out, balance_score, treated_mask = model.forward(
+            tf, treatment_index=treatment_idx)
+        loss = torch.mean(treated_mask *
+                          torch.square(tf.y - out.squeeze(-1))) + balance_score
```

Comment:
why are we just learning from positive `T`?

Reply:
That's the evaluation method in the [original paper](https://pdf.sciencedirectassets.com/272371/1-s2.0-S1532046423X00033/1-s2.0-S1532046423000606/main.pdf?X-Amz-Security-Token=IQoJb3JpZ2luX2VjEE4aCXVzLWVhc3QtMSJHMEUCIQCt7f9cbegQl8yVL1xlcTAC8c%2FGAN%2BM6UDPVTEQa6k%2BtgIgB4kp9oSB3gmhU11Bjw%2FFthB0HsnJbqd4lCmz9Enb7aMqvAUIh%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAFGgwwNTkwMDM1NDY4NjUiDIp5d0lhZXcrZu%2FkGSqQBUNbcfwkkQ4u0T711MiOCb3sOR8PJnMQDnXjAizjynm%2Fkp6kiGu2CY%2B6rtubNGjc4nFutHky1AeQYfHaCRtZ4%2Btk4EDUZ0ZTq14nUiE%2B0IHTNYIc4UK5rNfWU0wLsclcLAVKLKCCpQnBKmD%2F3z42zHkbZDSlm4HUA%2Fa1ZKJaj3nKp357XxgbgKfJcUXOYb8wWTd9LXg1N5iW0lgW6iB3lJ2qyDgeDLA83yCpYWaVZIZS9wztfA1%2B9XgwW9ga67TUrHW0wo96uyFhf2ZMOMBBMIByVKG0C5ojCHW%2BQRv4VAjsNeNGVCqOiXGt7tPM9qgAco7azdfx0G0GCgfMyPZ6HiSvjAxydH0ZsAJDrsFg0geGTx%2BsQfMB1ogpOX35zbijw3m716i4X97kP%2FHAj%2BNqDiu7UIw2fsQI5kTuQW93eVhXtM6Vw1H%2ByO%2F2Lw3QTKkta1px9VCn5%2BqLexX%2BKsYY%2BlDMDvjlyaTjLj%2Fz7t4GYUycbvP7uMPbOSmfVB3BkhzlblLHC62VlgkSTwN2ZOF41waNqVBUUylkwahloPpqrJBzFfCSAREG%2FSohj9OB9tmqxsP8%2BF1EL4E2OYuNKRCGpkTnOvJ%2F6tRdgldwuVvIODk3jmvFS0m84XYZJFI6UZQQgxwDGjksozogCveDQC2EGvMqt5ftxHa53kyeaECfO1p5HFtZ8iy%2BzliZEMFq5B0CLNVWgpralESOaukGhADZAtb4IjF7UdsIUQVkp6dU%2F0fjqGG6f6gwahf5mCqLzXEPqf5VF1G7QurgMOu6EmMNo1bCoE8LFh2bczEVGaer2FRCLNWcUp%2B71Wg%2B9hyOS397hJyYIJowCJTTcVNN8S3A%2Be5A9UMn7ZxQx7TRRTzyiIsBMPjP7bAGOrEBFlC%2ByT6kb0HCwmAMpJqNVsE4naZmiS4D%2F4jVdbqXeslSIqCoPHy6q6EvzOfsCadFxMfKSC0P2QOcrcdHRYXj1rpRECSWNkIoVe1H7nAk2bdSYIbxOJ9zWZwgy87zK4WkQPem9ZEqyPkq7j8uHGj1FyFkPgzEJXy1nXTYSFM%2BV2U5xUG2DbvN%2Bp57sALYu7kSX94cLyGsgAwtKG8L9%2BM4zWxjcUzwWf59S3KbDErKVepH&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20240414T063837Z&X-Amz-SignedHeaders=host&X-Amz-Expires=299&X-Amz-Credential=ASIAQ3PHCVTY4MTLC5BW%2F20240414%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Signature=904128f9f76d3383a0f7c4cac5b31eee6bcbc24a93184446348f62b5a18570e8&hash=f603074287768d9501c4b737bb71309e50abaaf6dd8d2bacb5c12c7aa88b40a2&host=68042c943591013ac2b2430a89b270f6af2c76d8dfd086a07176afe7c76c2c61&pii=S1532046423000606&tid=spdf-50b27d96-4339-481b-afcf-b76f3c3af3a7&sid=91f31f446a140645873a95a-ed1e7fb31c73gxrqa&type=client&tsoh=d3d3LnNjaWVuY2VkaXJlY3QuY29t&ua=1315585655500d5f0506&rr=8741b1ab7bf896dd&cc=us). 
<img width="318" alt="Screenshot 2024-04-15 at 1 05 41 AM" src="https://github.com/pyg-team/pytorch-frame/assets/22352182/1bf2807b-0403-4a4c-a496-d40519177584">
As for why, I need to look deeper

Reply:
They are evaluating on the original treated group

Reply:
Nvm I was looking at the wrong thing. You are right let me adjust the score